bigframes/core/groupby/__init__.py

```diff
@@ -326,7 +326,9 @@ def _agg_list(self, func: typing.Sequence) -> df.DataFrame:
             for f in func
         ]
         column_labels = [
-            (col_id, f) for col_id in self._aggregated_columns() for f in func
+            (self._block.col_id_to_label[col_id], f)
+            for col_id in self._aggregated_columns()
+            for f in func
         ]
```

Comment:
Can you also test with column multi-index? Not sure this will work for that case

Reply:
It works: https://screenshot.googleplex.com/BPBX6WpPFuXTuHs

Reply:
Not quite, we should have a case where the original columns have multi-part names

Reply:
Good catch! I've corrected the handling of multi-indexes when applying multiple aggregation functions.